examples/dqfd_atari.py

```diff
@@ -0,0 +1,101 @@
+import keras.backend as K
+from rl.agents.dqfd import DQfDAgent
+from rl.policy import LinearAnnealedPolicy, EpsGreedyQPolicy
+from rl.memory import PartitionedMemory
+from rl.core import Processor
+from rl.callbacks import TrainEpisodeLogger, ModelIntervalCheckpoint
+from rl.util import load_demo_data_from_file
+
+#We downsize the atari frame to 84 x 84 and feed the model 4 frames at a time for
+#a sense of direction and speed.
+INPUT_SHAPE = (84, 84)
+WINDOW_LENGTH = 4
+
+#Standard Atari processing
+class AtariDQfDProcessor(Processor):
+    def process_observation(self, observation):
+        assert observation.ndim == 3
+        img = Image.fromarray(observation)
+        img = img.resize(INPUT_SHAPE).convert('L')  # resize and convert to grayscale
+        processed_observation = np.array(img)
+        assert processed_observation.shape == INPUT_SHAPE
+        return processed_observation.astype('uint8')  # saves storage in experience memory
+
+    def process_state_batch(self, batch):
+        processed_batch = batch.astype('float32') / 255.
+        return processed_batch
+
+    def process_reward(self, reward):
+        return np.clip(reward, -1., 1.)
+        #return np.sign(reward) * np.log(1 + abs(reward)) #(used in paper)
+
+    def process_demo_data(self, demo_data):
+        #Important addition from dqn example
+        for step in demo_data:
+            step[0] = self.process_observation(step[0])
+            step[2] = self.process_reward(step[2])
+        return demo_data
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--mode', choices=['train', 'test'], default='train')
+parser.add_argument('--env-name', type=str, default='GopherDeterministic-v4')
+parser.add_argument('--weights', type=str, default=None)
+args = parser.parse_args()
+
+# Get the environment and extract the number of actions.
+env = gym.make(args.env_name)
+np.random.seed(231)
+env.seed(123)
+nb_actions = env.action_space.n
+print("NUMBER OF ACTIONS: " + str(nb_actions))
+
+#Standard DQN model architecture + l2 regularization to prevent overfitting on small demo sets.
+input_shape = (WINDOW_LENGTH, INPUT_SHAPE[0], INPUT_SHAPE[1])
+frame = Input(shape=(input_shape))
+cv1 = Conv2D(32, kernel_size=(8,8), strides=4, activation='relu', kernel_regularizer=l2(1e-4), data_format='channels_first')(frame)
+cv2 = Conv2D(64, kernel_size=(4,4), strides=2, activation='relu', kernel_regularizer=l2(1e-4), data_format='channels_first')(cv1)
+cv3 = Conv2D(64, kernel_size=(3,3), strides=1, activation='relu', kernel_regularizer=l2(1e-4), data_format='channels_first')(cv2)
+dense= Flatten()(cv3)
+dense = Dense(512, activation='relu', kernel_regularizer=l2(1e-4))(dense)
+buttons = Dense(nb_actions, activation='linear', kernel_regularizer=l2(1e-4))(dense)
+model = Model(inputs=frame,outputs=buttons)
+print(model.summary())
+
+processor = AtariDQfDProcessor()
+expert_demo_data = processor.process_demo_data(load_demo_data_from_file('gopher_expert_5k.npy'))
```

Comment:
I thought it would be too big to include. You can create it by uncommenting the line before it: `record_demo_data('GopherDeterministic-v4', steps=50000, data_filepath='gopher_expert.npy', frame_delay=0.03)`
(although that requires you to sit there and play the game for 50k steps).

I guess it comes down to whether we need the examples to be runnable out of the box, or if they can just serve as a template for people's own use cases.

Reply:
to clarify, that would be the line before it if you were looking at the most recent commit.